prog/weaver/main.go

```diff
@@ -35,6 +36,31 @@ type dnsConfig struct {
 	EffectiveListenAddress string
 }
 
+const (
+	versionCheckPeriod = 6 * time.Hour
+)
+
+func check() {
+	handleResponse := func(r *checkpoint.CheckResponse, err error) {
+		if err != nil {
+			Log.Printf("Error checking version: %v", err)
+		} else if r.Outdated {
+			Log.Printf("Weave version %s is available; please update at %s",
+				r.CurrentVersion, r.CurrentDownloadURL)
+		}
+	}
+
+	// Start background version checking
+	params := checkpoint.CheckParams{
+		Product:       "weave-net",
+		Version:       version,
+		SignatureFile: "",
```

Comment:
This is not going to do what you want when combined with weaveworks/go-checkpoint#1, as the call to `getSystemdUUID` is protected by an `if p.Signature == "" && p.SignatureFile != "" {` clause.